exercises/practice/nucleotide-count/.approaches/groupby-plus/content.md

```diff
@@ -0,0 +1,36 @@
+# `groupBy` with `+`
+
+```kotlin
+class Dna(val sequence: String) {
+    init {
+        require(sequence.all { it in "ACGT" })
+    }
+    val nucleotideCounts
+        get() = mapOf('A' to 0, 'C' to 0, 'G' to 0, 'T' to 0) + sequence.groupingBy { it }.eachCount()
+}
+```
+
+## Verification
+We receive a sequence as input, which we have to verificate.
+Because the sequence is part of the primary constructor, which can't contain runnable code, we have to place the check in an [_initializer block_][stdlib-constructors].
```

Comment:
Maybe add the relevant code after this bit of text? That will ensure that the reader doesn't have to scroll up to look at the relevant code bit:

````suggestion
Because the sequence is part of the primary constructor, which can't contain runnable code, we have to place the check in an [_initializer block_][stdlib-constructors].

```kotlin
init {
    require(sequence.all { it in "ACGT" })
}
```
````

This same comment applies to the following to sub-sections.